test/elm/interval/data.cql

```diff
@@ -51,47 +51,55 @@ define ContainsQuantity: Interval[1 'mg', 5 'mg'] contains 3 'mg'
+define NotContainsDayOfDateLowEdge: PrecisionDateIvlHighOpen contains day of DateTime(2012, 3, 1, 23, 59, 59, 999)
+define NotContainsDayOfDateBeyondHighEdge: PrecisionDateIvlHighOpen contains day of DateTime(2012, 9, 3, 0, 0, 0, 0)
+define ContainsDayOfDateImpreciseLowEdge: PrecisionDateIvlHighOpen contains day of DateTime(2012, 3, 2)
+define NotContainsDayOfDateImpreciseHighEdgeOpen: PrecisionDateIvlHighOpen contains day of DateTime(2012, 9, 2)
+define ContainsDayOfDateImpreciseHighEdgeClosed: PrecisionDateIvlHighClosed contains day of DateTime(2012, 9, 2)
+define ContainsDayOfDateVeryImpreciseMiddle: PrecisionDateIvlHighOpen contains day of DateTime(2012, 6)
+define NotContainsDayOfDateVeryImpreciseLow: PrecisionDateIvlHighOpen contains day of DateTime(2012, 2)
+define NotContainsDayOfDateVeryImpreciseHighEdgeOpen: PrecisionDateIvlHighOpen contains day of DateTime(2012, 10)
+define MayContainDayOfDateVeryImpreciseLow: PrecisionDateIvlHighOpen contains day of DateTime(2012, 3)
+define MayContainDayOfDateVeryImpreciseHigh: PrecisionDateIvlHighOpen contains day of DateTime(2012, 9)
+define MayContainDayOfDateVeryImpreciseSurrounding: PrecisionDateIvlHighOpen contains day of DateTime(2012)
 define ImpDateIvl: Interval[DateTime(2012, 3), DateTime(2012, 8)]
 define ImpreciseContainsDate: ImpDateIvl contains DateTime(2012, 6, 1, 0, 0, 0, 0)
 define ImpreciseNotContainsDate: ImpDateIvl contains DateTime(2012, 9, 1, 0, 0, 0, 0)
 define ImpreciseMayContainDate: ImpDateIvl contains DateTime(2012, 3, 15, 0, 0, 0, 0)
 define NegInfBegContainsInt: Interval[null, 5] contains -7
 define NegInfBegNotContainsInt: Interval[null, 5] contains 7
-define UnknownBegContainsInt: Interval(null, 5] contains 5
+define unknownOpenBegMayContainInt: Interval(null, 5] contains 5
+define unknownClosedBegContainsInt: Interval[null, 5] contains 5
 define UnknownBegMayContainInt: Interval(null, 5] contains -7
 define UnknownBegNotContainsInt: Interval(null, 5] contains 7
 define PosInfEndContainsInt: Interval[0, null] contains 123456789
 define PosInfEndNotContainsInt: Interval[0, null] contains -1
-define UnknownEndContainsInt: Interval[0, null) contains 0
+define UnknownOpenEndMayContainInt: Interval[0, null) contains 0
+define UnknownClosedEndContainsInt: Interval[0, null] contains 0
 define UnknownEndMayContainInt: Interval[0, null) contains 123456789
 define UnknownEndNotContainsInt: Interval[0, null) contains -1
 define NegInfBegContainsDate: Interval[null, DateTime(2013, 1, 1, 0, 0, 0, 0)) contains DateTime(1, 1, 1, 0, 0, 0, 0)
 define NegInfBegNotContainsDate: Interval[null, DateTime(2013, 1, 1, 0, 0, 0, 0)) contains DateTime(2013, 1, 2, 0, 0, 0, 0)
-define UnknownBegContainsDate: Interval(null, DateTime(2013, 1, 1, 0, 0, 0, 0)) contains DateTime(2012, 12, 31, 23, 59, 59, 999)
+define UnknownOpenBegMayContainDate: Interval(null, DateTime(2013, 1, 1, 0, 0, 0, 0)) contains DateTime(2012, 12, 31, 23, 59, 59, 999)
```

Comment:
This one's more tricky, but in theory, given the `ms` precision of the system, _maybe_ this should be `true`?  Less sure on this one.  It depends on if a completely empty interval is possible in CQL (and I really don't know).  In other words, would this be allowed?
```
(DateTime(2013, 1, 1, 0, 0, 0, 0), DateTime(2013, 1, 1, 0, 0, 0, 0))
```
Given that both boundaries are open and both values are equal, this would be an interval containing _no points_.  If that's allowed, then I guess the answer to this specific test should be `null` (unknown), as it currently is.

Might be another good question for Bryn.